ospec.js

```diff
@@ -166,7 +166,12 @@ else window.o = m()
 				}
 				if (fn.length > 0) {
 					var body = fn.toString()
-					arg = (body.match(/\((.+?)(?:\s|\/\*[\s\S]*?\*\/|\/\/.*?\n)*[,\)].*\s=>/) || body.match(/^(.+?)(?:\s|\/\*[\s\S]*?\*\/|\/\/.*?\n)*=>/) || body.match(/\((?:\s|\/\*[\s\S]*?\*\/|\/\/.*?\n)*(.+?)(?:\s|\/\*[\s\S]*?\*\/|\/\/.*?\n)*[,\)]/) || []).pop()
+					arg = (
+						body.match(/\((.+?)(?:\s|\/\*[^]*?\*\/|\/\/[^]*?\n)*[,\)].*\s=>/)
```

Comment:
it maybe doesn't accept `(done)=>` with no whitespace

it will be better to 
- remove the 1st arrow-function matcher
- skip 2nd arrow-function matcher if top of `body` is "(" like this,   
```
body.match(/^([^\(].*?)(?:\s|\/\*[^]*?\*\/|\/\/[^]*?\n)*=>/)
```

Reply:
This is a mess...

I'm going to redo these from the ground up using `compose-regexp`, hoefully it'll add some clarity.

Reply:
See #16 hopefully this one is reliable...

Reply:
#16 parser looks good, thank you!! 😄